### DISCUSSION TITLE
Getting a warning - The discovered document is not of type "TadaDocumentNode". when trying to use turbo

### DISCUSSION THREAD

**Discussion by anirudhsama** [0no-co/gql.tada#244] [Q&A] [answered]
Hi, 

I ran `gql-tada turbo` and got the above error and the cache file generated was just:

```
/* eslint-disable */
/* prettier-ignore */
import type { TadaDocumentNode, $tada } from 'gql.tada';

declare module 'gql.tada' {
 interface setupCache {

  }
}
```

Is this expected?

**Comment by kitten** [comment]
This typically means that the `graphql()` call that's being targeted hasn't actually evaluated to a `TadaDocumentNode` type.

Check the warning closely that `turbo` gives you, go to that line, and check the type hint in your editor to see what type the return value of `graphql()` is at that place.

**Reply by anirudhsama** [reply]
It's showing `TadaDocumentNode` in the tooltip, but as `DocumentNode` (from `@no-co/graphql.web`) in the documentation. See screenshot. The `graphql` function definition is as follows:

```
export const graphql = initGraphQLTada<{
	introspection: introspection;
	scalars: {
		timestamp: string;
		timestamptz: string;
		date: string;
		uuid: string;
		jsonb: any;
		json: any;
	};
}>();
```

![CleanShot 2024-04-22 at 18 25 02@2x](https://github.com/0no-co/gql.tada/assets/832760/dd42bdf8-61d3-4f10-bab5-a5105e77dd4a)

Is some configuration incorrect?

**Reply by anirudhsama** [reply]
The variable is also being typed as `TadaDocumentNode`

![CleanShot 2024-04-22 at 18 29 42@2x](https://github.com/0no-co/gql.tada/assets/832760/c1d0832d-9060-4823-807e-95e9fd79375e)

**Reply by kitten** [reply, marked as answer]
If you did validate that this is the line that `turbo` points to, then I'm not sure what the issue may be, but without a reproduction it'll be hard for me to check 🤔 

Edit:
also, re. this:
> but as DocumentNode (from @no-co/graphql.web) in the documentation

that's just documentation. Ultimately, for the TSDocs the remark there is trying to make it clear that `DocumentNode` is the relevant return type. So, entirely unrelated

**Reply by anirudhsama** [reply]
Alright. Let me try to reproduce in a repo and ping.

**Reply by anirudhsama** [reply]
Tried the turbo command after updating to 1.6.2 and it worked. So not sure what the error was. Anyway, fixed now! Thanks.

**Reply by scamden** [reply]
hmm i'm getting this warning on v1.8.10. Hovering the call site shows a TadaDocumentNode and it typechecks fine. But turbo warns for every call site and doesn't add anything to the cache file. 

FWIW i'm in a mono repo and the tada setup is all within one of the workspaces.

**Reply by dennemark** [reply]
same here. i am using pnpm.

**Reply by scamden** [reply]
I'm also using pnpm fwiw, any thoughts on this @kitten ?

**Reply by scamden** [reply]
long time here but I found that this seems to have something to do with not processing tsconfig.json extends properly. Tsconfig now accepts node package references in that property eg:

`"extends": "@vtp/rig/base-tsconfig.json",`

for me that is a local pnpm workspace to share configs. 

if i do this instead:

`"extends": "./node_modules/@vtp/rig/base-tsconfig.json",`

i no longer get the warning. @kitten any chance you could fix the extends parsing now that I have a cause figured out? (it's either a failure to process node package references here altogether or a failure to process them when they are local pnpm workspace packages, not sure which.)

(The specific reason it was failing in my case was moduleResolution: 'nodenext' needed to be set but wasn't, but I imagine you want to properly honor all of the tsconfig settings to avoid problems. )

**Reply by kitten** [reply]
That'd be rather odd given: https://github.com/0no-co/gql.tada/blob/1c77ae8502a5dd12b82aee71749d8e2b7226ffbc/packages/cli-utils/src/ts/factory.ts#L74

Which will force the correct resolution  option for evaluation since it had little consequence and that'll always work.

We might have an error that's not being read out properly or another edge case, but without a reproduction I probably won't look into this, since it's a configuration issue and generally I don't hunt down configuration issues without repros 😅

**Reply by scamden** [reply]
I'll do my best to make a repro repo. 

FWIW i just double checked. If I set extends back to a non-relative path I see the warning. If I then add `"moduleResolution": "nodenext",` to compilerOptions the warning goes away. Can't explain why but that's what's happening.

**Reply by scamden** [reply]
OK i got the repro and it looks like it's due to sub path imports, the moduleResolution seems to fix it for some reason. 

https://github.com/scamden/tada-repro

**Reply by scamden** [reply]
@kitten relevant lines in the repro:

the sub path import of tada:
https://github.com/scamden/tada-repro/blob/dbba3e1a90b6e791e4681998812da4851d4843a4/packages/app/src/index.ts#L1

making this a relative path fixes it because it pulls in moduleResolution settings (not sure why package name reference in extends fails):
https://github.com/scamden/tada-repro/blob/dbba3e1a90b6e791e4681998812da4851d4843a4/packages/app/tsconfig.json#L2

set moduleResolution directly without changing extends to prove this setting alone somehow fixes it
https://github.com/scamden/tada-repro/blob/dbba3e1a90b6e791e4681998812da4851d4843a4/packages/app/tsconfig.json#L5

**Reply by scamden** [reply]
@kitten went through some effort to repro this, would it make sense to open an issue to get this fixed?

**Reply by scamden** [reply]
any.. update.. here..
